File: 프로그래머스/1/68644. 두 개 뽑아서 더하기/두 개 뽑아서 더하기.py
# sorted 함수는 기본적으로 리스트, 튜플, 문자열, 집합 등 반복 가능한 객체를 정렬하여 새로운 정렬된 리스트를 반환하는 함수입니다. 기본적으로 오름차순 정렬 

def solution(numbers):
    answer = []
    
    for i in range(len(numbers)):
        for j in range(i+1, len(numbers)):
            answer.append(numbers[i] + numbers[j])
    return sorted(list(set(answer)))

File: 프로그래머스/1/68644. 두 개 뽑아서 더하기/두 개 뽑아서 더하기.java
import java.util.HashSet;

class Solution {
    public int[] solution(int[] numbers){
        // 1. 중복값 제거를 위해 해시셋 선택 - 어차피 나중에 sorted()로 순서 다시 정리해야
        // HashSet vs LinkedHashSet vs TreeSet
        // HashSet 은 평균적으로 O(1). 데이터의 유일성만 중요, 순서가 중요X
        // LinkedHashSet 은 평균적으로 O(1). 데이터 유일성 + 순서 유지
        // TreeSet 은 평균적으로  O(logN). 데이터 순서 유지 + 집합 특성 유지할 때
        HashSet<Integer> set = new HashSet<>();

        // 두 수를
        for (int i=0; i<numbers.length-1; i++){
            for (int j = i+1; j < numbers.length; j++){
                set.add(numbers[i] + numbers[j]);
            }
        }

        //
        return set.stream().sorted().mapToInt(Integer::intValue).toArray();
    }
}